AutoPBW.WPF/MainWindow.xaml.cs

```diff
@@ -155,167 +155,175 @@ private void RefreshData()
 			// set cursor
 			Cursor = Cursors.Wait;
 
-			// remember selection
-			var selGame = gridPlayerGames.SelectedItem as PlayerGame;
-			var selCode = selGame == null ? null : selGame.Code;
-			var selPlayerNumber = selGame == null ? null : (int?)selGame.PlayerNumber;
-			IEnumerable<HostGame> hostGames = null;
-			IEnumerable<PlayerGame> playerGames = null;
+			if (!PBW.isLoggedIn)
+			{
+				Login();
+			}
 
-			try
+			if (PBW.isLoggedIn)
 			{

```

Comment:
Github can't diff indent changes well - these additions are all that's changed in this file apart from a closing brace later, and a lot of indented lines.